specfile/specfile.py

```diff
@@ -120,18 +183,32 @@ def _dump_debug_info(self, message) -> None:
         )
 
     @staticmethod
-    def _read_lines(path: Path) -> Tuple[List[str], bool]:
-        content = path.read_text(encoding="utf8", errors="surrogateescape")
-        return content.splitlines(), content[-1] == "\n"
+    def _read_lines(stream: TextIO) -> Tuple[List[str], bool]:
+        stream.seek(0)
+        content = stream.read()
+        return content.splitlines(), content.endswith("\n")
+
+    @property
+    def is_writable(self) -> bool:
+        return hasattr(self._file, "write") and callable(
+            getattr(self._file, "write", None)
+        )
 
     @property
     def path(self) -> Path:
         """Path to the spec file."""
-        return self._path
+        if hasattr(self._file, "name"):
+            return Path(self._file.name)
+        return None
 
     @path.setter
     def path(self, value: Union[Path, str]) -> None:
-        self._path = Path(value)
+        self._file.close()
+        new_path = Path(value)
+        self._file = new_path.open("r+", encoding="utf8", errors="surrogateescape")
```

Comment:
Since the same arguments are used on multiple places, it would be nice to make them a constant, so they can be changed at a single location (if needed).